draft-ietf-quic-transport.md

```diff
@@ -735,20 +735,15 @@ discarded upon receipt.
 
 A STOP_SENDING frame requests that the receiving endpoint send a RESET_STREAM
 frame.  An endpoint that receives a STOP_SENDING frame MUST send a RESET_STREAM
-frame if the stream is in the Ready or Send state.  If the stream is in the Data
-Sent state and any outstanding data is declared lost, an endpoint SHOULD send a
-RESET_STREAM frame in lieu of a retransmission.
+frame if the stream is in the Ready or Send state.  If the stream is in the
+"Data Sent" state and any outstanding data is declared lost, an endpoint SHOULD
+send a RESET_STREAM frame in lieu of a retransmission.
 
 An endpoint SHOULD copy the error code from the STOP_SENDING frame to the
 RESET_STREAM frame it sends, but MAY use any application error code.  The
 endpoint that sends a STOP_SENDING frame MAY ignore the error code carried in
 any RESET_STREAM frame it receives.
 
-If the STOP_SENDING frame is received on a stream that is already in the
-"Data Sent" state, an endpoint that wishes to cease retransmission of
-previously-sent STREAM frames on that stream MUST first send a RESET_STREAM
-frame.
```

Comment:
They are very slightly different -- this para says that an endpoint that wants to stop retransmitting MUST send a RESET_STREAM first. The earlier para says that a RESET_STREAM is sent instead of a retransmission.

That said, I think this is basically redundant. I would replace the language in the earlier para with the one from here, which is more accurate. We don't need the endpoint to send RESET_STREAM every time it detects a loss. Only that it send a RESET_STREAM before stopping retrnamissions.

Reply:
I think the real point is that you don't have to send a RESET_STREAM immediately if you've already sent everything and closed the stream, but you SHOULD send a RESET_STREAM if any data later gets lost.  See if this new text works better.

Reply:
I like the new text.  Thanks.